proposals/sidepanel-close-api.md

```diff
@@ -0,0 +1,132 @@
+# New close() Method for the sidePanel API
+
+**Summary**
+Proposal to introduce a programmatic close method for the side panel API.
+
+**Document Metadata**
+
+**Author:** [hharshas](https://github.com/hharshas)
+
+**Sponsoring Browser:** Chromium
+
+**Contributors:** [hharshas](https://github.com/hharshas) (
+  [Summer Of Code](https://summerofcode.withgoogle.com/)), mentor: oliverdunk, solomonkinard
+
+**Created:** 2025-05-27
+
+**Related Issues:** [#521](https://github.com/w3c/webextensions/issues/521),
+ [#chromium](https://issues.chromium.org/issues/403765214)
+
+## Motivation
+
+### Objective
+
+What does this API enable?
+- sidePanel.close() method will allow the sidebar to close programmatically.
+
+Why do we need it?
+- Currently, extensions cannot directly close their own sidebar from the
+ background script, 
+developers have to:
+  1. Send a message from a background script to the side panel.
+  2. Call window.close() from the side panel window on message delivery.
```

Comment:
Fixes a rendering bug where "developers have to:" appeared as a paragraph separating the unordered list that precedes it from the ordered list that follows.

Also styles "window.close()" as monospace to signal this text is a snippet of code.

```suggestion
Why do we need it?
- Currently, extensions cannot directly close their own sidebar from the
  background script, developers have to:
  1. Send a message from a background script to the side panel.
  2. Call `window.close()` from the side panel window on message delivery.
```